File: LeetCode/LeetCode_Easy/TypeScript/BinarySearch.ts
function search(nums: number[], target: number): number {
  let left = 0;
  let right = nums.length - 1;

  while (left <= right) {
    const half = Math.floor((left + right) / 2);

    if (nums[half] === target) {
      return half;
    } else if (nums[half] < target) {
      left = half + 1;
    } else {
      right = half - 1;
    }
  }

  return -1;
}

File: LeetCode/LeetCode_Easy/C++/BinarySearch.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;

        while (left <= right) {
            int half = floor((left + right) / 2);
            
            if (nums[half] == target) {
                return half;
            }

            if (nums[half] < target) {
                left = half + 1;
            } else {
                right = half - 1;
            }
        }

        return -1;
    }
};